expect/expect.ts

```diff
@@ -79,8 +79,7 @@ export interface Expected {
   toReturnTimes(expected: number): void;
   toReturnWith(expected: unknown): void;
   toStrictEqual(candidate: unknown): void;
-  // deno-lint-ignore no-explicit-any
-  toThrow<E extends Error = Error>(expected?: new (...args: any[]) => E): void;
+  toThrow<E extends Error = Error>(expected?: unknown): void;
```

Comment:
This is a change to the public API, however it only makes the parameter type wider so shouldn't break any backwards compatibility. I could provide a more correct type here, but the convention seems to be that any complex types in the expect API are just replaced with unknown. The proper type would be something like:
```typescript
interface Expect {
  // deno-lint-ignore no-explicit-any
  toThrow<E extends Error = Error>(expected?: string | RegExp | E | (new (...args: any[]) => E)): void
}
```
Happy to do this if you think its better

Reply:
The above type in your comment makes more sense to me. Can you update the typing to it?